src/azul/project/hca/metadata_generator.py

```diff
@@ -0,0 +1,197 @@
+            "^\\*\\.file_core\\.file_format",
+            "^sequence_file.*",
+            "^analysis_file.*",
+            "^donor_organism.*",
+            "^specimen_from_organism.*",
+            "^cell_suspension.*",
+            "^.*protocol.*",
+            "^project.",
+            "^analysis_process.*",
+            "^process.*",
+            "^bundle_.*",
+            "^\\*\\.provenance\\.update_date"
+        ]
+
+        self.default_ignore = ignore if ignore else [
+            "describedBy",
+            "schema_type",
+            "submission_date",
+            "update_date",
+            "biomaterial_id",
+            "process_id",
+            "contributors",
+            "publications",
+            "protocol_id",
+            "project_description",
+            "file_format",
+            "file_name"
+        ]
+
+        self.default_format_filter = format_filter
+        # TODO temp until block filetype is needed
+        self.default_blocked_file_ext = {'csv', 'txt', 'pdf'}
+
+    def _get_file_uuids_from_objects(self, list_of_metadata_objects: List[JSON]) -> JSON:
+        file_uuids = {}
+        for _object in list_of_metadata_objects:
+            if "schema_type" not in _object:
+                raise MissingSchemaTypeError("JSON objects must declare a schema type")
+
+            if _object["schema_type"] == "file" and self._deep_get(_object, ["provenance", "document_id"]):
+                file_uuid = self._deep_get(_object, ["provenance", "document_id"])
+                file_uuids[file_uuid] = _object
+
+        if not file_uuids:
+            raise MissingFileTypeError("Bundle contains no data files")
+
+        return file_uuids
+
+    def _deep_get(self, d: JSON, keys: List):
+        if not keys or d is None:
+            return d
+        return self._deep_get(d.get(keys[0]), keys[1:])
+
+    @staticmethod
+    def _set_value(master: JSON, key: str, value: Any) -> None:
+
+        if key not in master:
+            master[key] = str(value)
+        else:
+            existing_values = master[key].split("||")
+            existing_values.append(str(value))
+            uniq = sorted(list(set(existing_values)))
+            master[key] = "||".join(uniq)
+
+    def _flatten(self, master: JSON, obj: JSON, parent: str) -> None:
+        for key, value in obj.items():
+            if key in self.default_ignore:
+                continue
+
+            newkey = parent + "." + key
+            if isinstance(value, dict):
+                self._flatten(master, obj[key], newkey)
+            elif isinstance(value, list):
+                for item in value:
+                    if isinstance(item, dict):
+                        self._flatten(master, item, newkey)
+                    else:
+                        self._set_value(master, newkey, item)
+            else:
+                self._set_value(master, newkey, value)
+
+    @staticmethod
+    def _get_schema_name_from_object(_object: JSON):
```

Comment:
Convention is to use `object_`, not `_object`.

https://www.python.org/dev/peps/pep-0008/#descriptive-naming-styles, under `single_trailing_underscore_`

But: Now that you touched this, I would actually prefer `obj` instead of `object_` or `_object`